lib/chef/provider/package.rb

```diff
@@ -464,10 +464,7 @@ def current_version_array
 
       # @return [Array] new_version(s) as an array
       def new_version_array
-        @new_version_array ||=
-            [ new_resource.version ].flatten.map do |v|
-              ( v.nil? || v.empty? ) ? nil : v
-            end
+        [ new_resource.version ].flatten
```

Comment:
@lamont-granquist  did this, as I recall, because we can get both nil or `''`. For exmaple, we support:

```
package ['one', 'two', 'three'] do
  version ['1.2', '', '4']
  action :install
end
```

And this meant we didn't have to check for both everywhere...